File: python-pr/52/python/code-quality-readability-maintainability/cache_manager_19.py
# {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
class CacheManager:
    def __init__(self, max_size=100):
        self.cache = {}
        self.max_size = max_size
        self.access_count = {}
    def get(self, key):
        if key in self.cache:
            self.access_count[key] = self.access_count.get(key, 0) + 1
            return self.cache[key]
        return None
    def set(self, key, value):
        if len(self.cache) >= self.max_size:
            least_used = min(self.access_count, key=self.access_count.get)
            del self.cache[least_used]
            del self.access_count[least_used]
        self.cache[key] = value
        self.access_count[key] = 1
# {/fact}

File: js-pr/56/javascript/code-quality-readability-maintainability/cache_manager_19.js
// {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
class CacheManager {
    constructor(maxSize = 100) {
        this.cache = new Map();
        this.maxSize = maxSize;
        this.accessCount = new Map();
    }
    get(key) {
        if (this.cache.has(key)) {
            this.accessCount.set(key, (this.accessCount.get(key) || 0) + 1);
            return this.cache.get(key);
        }
        return null;
    }
    set(key, value) {
        if (this.cache.size >= this.maxSize) {
            let leastUsed = [...this.accessCount.entries()].reduce((a, b) => a[1] < b[1] ? a : b)[0];
            this.cache.delete(leastUsed);
            this.accessCount.delete(leastUsed);
        }
        this.cache.set(key, value);
        this.accessCount.set(key, 1);
    }
}
// {/fact}